rfc/0003-tests-arch.md

```diff
@@ -0,0 +1,234 @@
+to integrations tests.[3](https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html)
+
+When implemented the developer should run all the test with a single command like:
+
+```bash
+# code code code
+make acceptance-tests
+# or
+make integration-tests
+```
+
+These tests should also run on every change proposed in the component
+repository (e.g. push, pull request).
+
+
+## End-to-end tests
+
+These are the most complex tests. They will simulate what Kubewarden users will
+do to install, upgrade, configure and possibly uninstall Kubewarden stack.
+
+In these tests, different components versions would be tested together.
+Therefore, ensuring that the user will not face issues when using different
+components configuration and versions together. This should ensure that
+Kubernetes operator will have the expected features working.
+
+These tests are trigger when we have new releases (or release candidate) of
+any of the Kubewarden stack.  It should follow some steps:
+
+1. Setup Kubernetes cluster
+  Kubewarden is expected to run in multiple Kubernetes versions and machine
+  architectures. So, the end-to-end tests should ensure that all the supported
+  versions and architectures are tested. The initial versions of the tests can
+  be simple in a single arch and version.
+2. Install Kuberwarden stack on the cluster from step 1
+  The version to be tested should be installed. This means that when a component
+  is released. It should be tested against all the other components versions which
+it is supposed to work together.
+3. Run basic end-to-end tests
+  Validate if the basic features are working as expected
+4. Run reconfiguration tests
+  Validates if the user is able to reconfigure a Kubewarden stack installed
+5. Run upgrade tests
+  Validate if the user is able to upgrade and continue to work with the latest release.
+
+Steps 3,4 and 5 may run in parallel when necessary (e.g. time to execute all tests
+is to long)
+
+Considering the tests will validate multiple components of the Kubewarden stack
+together (controller and Policy Server), it desirable to keep all the
+end-to-end test files in a separated repository. This repository should contains
+all the yaml files, scripts, and helper code to run the tests. This also helps
+to avoid duplicating tests that validate some feature that impacts multiple
+components.
+
+The end-to-end tests repository should be independent. Which means that it should
+run the tests independently of the tests and development cycle of the other
+Kubewarden repositories. But every time a new component release is available
+for testing, the end-to-end tests should be triggered automatically with no human
+intervention.  Besides that, running the end-to-end test on demand and in pre defined
+periods is also desirable.
+
+We could also configure the end-to-end tests to run against the latest version of
+all Kubewarden components. Thus, spotting some possible issues before a official
```

Comment:
I think we should also have a nightly job from all components in `main`, that can spot failures during the development cycle, acting as a canary.